### ISSUE TITLE
stack ghci sometimes loads old versions of modules

### ISSUE THREAD

**Issue by andrewthad** [commercialhaskell/stack#685]
Here's my situation. I have two files: `Application.hs` and `MyCron.hs`. If I run `stack ghci` and then:

    :l src/Application.hs
    import Application
    import MyCron

This all works fine. Since `Application` import `MyCron`, the module is correctly in scope and everything is good. But let's say that I had forgotten to export something from `MyCron`. So I amend it's export list:

    module MyCron 
      ( scheduleAllJobs
      , parseTor -- adding this one
      ) where

Now, I should be able to do something like this in GHCi:

    :reload
    import Application
    import MyCron

No luck. `parseTor` is not in scope. If I quit out of GHCi and then open it back up with `stack ghci` and then do the loading and importing, it still doesn't work. If I do that again and run `stack clean` before starting `stack ghci`, it still doesn't work. I have to run:

    stack clean
    stack build
    stack ghci

And finally, my newly exported function can come into scope. If I can clarify this issue further, please let me know.

**Comment by andrewthad** [comment]
@chrisdone This may have been self inflicted. I just noticed that I had this at the top of my .ghci file:

    :set -i.:config:dist/build/autogen

I don't know what the effect of that is yet, but tomorrow I'm going to see if removing it makes this issue go away.

**Comment by chrisdone** [comment]
I think `:reload` only works on the modules of `:load`, not `import`. These two are different concepts. So you're only reloading `Application`, nothing else. This is a regular GHCi thing. Closing unless you think there's still a bug.
